File: kt/task/17recur.kt
package task

class RecurTask {

    var datScanner = scanner.Scanner()
    var ansScanner = scanner.Scanner()

    fun solve(pathPrefix: String) {
        for (taskNo in 1..30) {
            var taskNoStr = "%03d".format(taskNo)
            var fileName: String
            for (testNo in 1..100) {
                var testNoStr = "%03d".format(testNo)
                fileName = "$pathPrefix/17recur/Recur$taskNoStr/$testNoStr/$testNoStr"
                if (!datScanner.changeReader("$fileName.dat")) {
                    println("Recur$taskNoStr couldn't open datScanner source")
                    return
                }
                if (!ansScanner.changeReader("$fileName.ans")) {
                    println("Recur$taskNoStr couldn't open ansScanner source")
                    return
                }
                val ok = when (taskNo) {
                    1 -> this.recur1()
                    2 -> this.recur2()
                    3 -> this.recur3()
                    4 -> this.recur4()
                    5 -> this.recur5()
                    6 -> this.recur6()
                    7 -> this.recur7()
                    8 -> this.recur8()
                    9 -> this.recur9()
                    10 -> this.recur10()
                    11 -> this.recur11()
                    12 -> this.recur12()
                    13 -> this.recur13()
                    14 -> this.recur14()
                    15 -> this.recur15()
                    16 -> this.recur16()
                    17 -> this.recur17()
                    18 -> this.recur18()
                    19 -> this.recur19()
                    20 -> this.recur20()
                    21 -> this.recur21()
                    22 -> this.recur22()
                    23 -> this.recur23()
                    24 -> this.recur24()
                    25 -> this.recur25()
                    26 -> this.recur26()
                    27 -> this.recur27()
                    28 -> this.recur28()
                    29 -> this.recur29()
                    30 -> this.recur30()
                    else -> false
                }
                scanner.removeFiles()
                if (!ok) {
                    println("Recur$taskNoStr Test#$testNoStr has not been tested")
                    return
                }
                if (!datScanner.eof()) {
                    println("Recur$taskNoStr Test#$testNoStr input is not enough")
                    return
                }
                if (!ansScanner.eof()) {
                    println("Recur$taskNoStr Test#$testNoStr output is not enough")
                    return
                }
            }
            println("Recur$taskNoStr has been tested!")
        }
    }

    private fun recur1(): Boolean {
        return false
    }

    private fun recur2(): Boolean {
        return false
    }

    private fun recur3(): Boolean {
        return false
    }

    private fun recur4(): Boolean {
        return false
    }

    private fun recur5(): Boolean {
        return false
    }

    private fun recur6(): Boolean {
        return false
    }

    private fun recur7(): Boolean {
        return false
    }

    private fun recur8(): Boolean {
        return false
    }

    private fun recur9(): Boolean {
        return false
    }

    private fun recur10(): Boolean {
        return false
    }

    private fun recur11(): Boolean {
        return false
    }

    private fun recur12(): Boolean {
        return false
    }

    private fun recur13(): Boolean {
        return false
    }

    private fun recur14(): Boolean {
        return false
    }

    private fun recur15(): Boolean {
        return false
    }

    private fun recur16(): Boolean {
        return false
    }

    private fun recur17(): Boolean {
        return false
    }

    private fun recur18(): Boolean {
        return false
    }

    private fun recur19(): Boolean {
        return false
    }

    private fun recur20(): Boolean {
        return false
    }

    private fun recur21(): Boolean {
        return false
    }

    private fun recur22(): Boolean {
        return false
    }

    private fun recur23(): Boolean {
        return false
    }

    private fun recur24(): Boolean {
        return false
    }

    private fun recur25(): Boolean {
        return false
    }

    private fun recur26(): Boolean {
        return false
    }

    private fun recur27(): Boolean {
        return false
    }

    private fun recur28(): Boolean {
        return false
    }

    private fun recur29(): Boolean {
        return false
    }

    private fun recur30(): Boolean {
        return false
    }
}

File: go/task/17recur.go
package task

import (
	"fmt"

	"github.com/pkg/errors"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/checker"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/proc"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/scanner"
)

type recurTask struct {
	dataPath string
	name     string
	count    int
	scanner  scanner.Scanner
	checker  checker.Checker
}

func NewRecurTask(pathPrefix string) Task {
	return &recurTask{
		dataPath: pathPrefix + "/17recur/Recur%03d/%03d/%03d",
		name:     "Recur",
		count:    30,
	}
}

func (rt *recurTask) GetCount() int { return rt.count }

func (rt *recurTask) GetName() string { return rt.name }

func (rt *recurTask) SetData(taskNo int, testNo int) error {
	var filePath = fmt.Sprintf(rt.dataPath, taskNo, testNo, testNo)
	scanner, err := scanner.New(filePath + ".dat")
	if err != nil {
		return errors.Wrap(err, "scanner.New")
	}
	checker, err := checker.New(filePath + ".ans")
	if err != nil {
		return errors.Wrap(err, "checker.New")
	}
	rt.scanner = scanner
	rt.checker = checker
	return nil
}

func (rt *recurTask) ScannerEOF() bool { return rt.scanner.EOF() }

func (rt *recurTask) CheckerEOF() bool { return rt.checker.EOF() }

func (rt *recurTask) CleanData() {
	rt.scanner.RemoveFiles()
	rt.checker.RemoveFiles()
}

func (rt *recurTask) GetFn(taskNo int) func() bool {
	switch taskNo {
	case 1:
		return rt.recur1
	case 2:
		return rt.recur2
	case 3:
		return rt.recur3
	case 4:
		return rt.recur4
	case 5:
		return rt.recur5
	case 6:
		return rt.recur6
	case 7:
		return rt.recur7
	case 8:
		return rt.recur8
	case 9:
		return rt.recur9
	case 10:
		return rt.recur10
	case 11:
		return rt.recur11
	case 12:
		return rt.recur12
	case 13:
		return rt.recur13
	case 14:
		return rt.recur14
	case 15:
		return rt.recur15
	case 16:
		return rt.recur16
	case 17:
		return rt.recur17
	case 18:
		return rt.recur18
	case 19:
		return rt.recur19
	case 20:
		return rt.recur20
	case 21:
		return rt.recur21
	case 22:
		return rt.recur22
	case 23:
		return rt.recur23
	case 24:
		return rt.recur24
	case 25:
		return rt.recur25
	case 26:
		return rt.recur26
	case 27:
		return rt.recur27
	case 28:
		return rt.recur28
	case 29:
		return rt.recur29
	case 30:
		return rt.recur30
	default:
		return nil
	}
}

// { "no": 1, "dat": "", "ans": "1" }
func (rt *recurTask) recur1() bool {
	for index := 0; index < 5; index++ {
		var number = rt.scanner.NextInt()
		var factorial = proc.Fact(number)
		if !rt.checker.CompareFloat64(1, factorial) {
			return false
		}
	}
	return true
}

// { "no": 2, "dat": "", "ans": "1" }
func (rt *recurTask) recur2() bool {
	return false
}

// { "no": 3, "dat": "", "ans": "E (OR 15)" }
func (rt *recurTask) recur3() bool {
	return false
}

// { "no": 4, "dat": "", "ans": "" }
func (rt *recurTask) recur4() bool {
	return false
}

// { "no": 5, "dat": "", "ans": "" }
func (rt *recurTask) recur5() bool {
	return false
}

// { "no": 6, "dat": "", "ans": "" }
func (rt *recurTask) recur6() bool {
	return false
}

// { "no": 7, "dat": "", "ans": "" }
func (rt *recurTask) recur7() bool {
	return false
}

// { "no": 8, "dat": "2", "ans": "8" }
func (rt *recurTask) recur8() bool {
	return false
}

// { "no": 9, "dat": "", "ans": "" }
func (rt *recurTask) recur9() bool {
	return false
}

// { "no": 10, "dat": "", "ans": "" }
func (rt *recurTask) recur10() bool {
	return false
}

// { "no": 11, "dat": "", "ans": "" }
func (rt *recurTask) recur11() bool {
	return false
}

// { "no": 12, "dat": "", "ans": "" }
func (rt *recurTask) recur12() bool {
	return false
}

// { "no": 13, "dat": "", "ans": "" }
func (rt *recurTask) recur13() bool {
	return false
}

// { "no": 14, "dat": "", "ans": "" }
func (rt *recurTask) recur14() bool {
	return false
}

// { "no": 15, "dat": "", "ans": "" }
func (rt *recurTask) recur15() bool {
	return false
}

// { "no": 16, "dat": "", "ans": "" }
func (rt *recurTask) recur16() bool {
	return false
}

// { "no": 17, "dat": "", "ans": "" }
func (rt *recurTask) recur17() bool {
	return false
}

// { "no": 18, "dat": "", "ans": "" }
func (rt *recurTask) recur18() bool {
	return false
}

// { "no": 19, "dat": "", "ans": "" }
func (rt *recurTask) recur19() bool {
	return false
}

// { "no": 20, "dat": "", "ans": "" }
func (rt *recurTask) recur20() bool {
	return false
}

// { "no": 21, "dat": "", "ans": "" }
func (rt *recurTask) recur21() bool {
	return false
}

// { "no": 22, "dat": "", "ans": "" }
func (rt *recurTask) recur22() bool {
	return false
}

// { "no": 23, "dat": "", "ans": "" }
func (rt *recurTask) recur23() bool {
	return false
}

// { "no": 24, "dat": "", "ans": "" }
func (rt *recurTask) recur24() bool {
	return false
}

// { "no": 25, "dat": "", "ans": "" }
func (rt *recurTask) recur25() bool {
	return false
}

// { "no": 26, "dat": "", "ans": "" }
func (rt *recurTask) recur26() bool {
	return false
}

// { "no": 27, "dat": "", "ans": "" }
func (rt *recurTask) recur27() bool {
	return false
}

// { "no": 28, "dat": "", "ans": "" }
func (rt *recurTask) recur28() bool {
	return false
}

// { "no": 29, "dat": "", "ans": "" }
func (rt *recurTask) recur29() bool {
	return false
}

// { "no": 30, "dat": "", "ans": "" }
func (rt *recurTask) recur30() bool {
	return false
}
